sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md

```diff
@@ -0,0 +1,38 @@
+## API Report File for "@azure/apimanagement-custom-widget-scaffolder"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+// @public
+export function generateProject(widgetConfig: TCustomWidgetBaseConfig, deployConfig: TDeployConfig, miscConfig?: TMiscConfig): Promise<void>;
```

Comment:
In general we try to organize our signatures as follows

```typescript
function foo(requiredParam1: FooParam, requiredParam2: BarParam, options?: ScaffoldingOptions)
```

Where all required parameters are positional and all optional parameters are inside an options bag.

I see that you have 4 required parameters. From those, are there any that would make sense to make optional? For example, default your Tech parameter to something like React and put it in an options bag?

```ts
export function scaffoldWidget(displayName: string, managementApiEndpoint: string, resourceId: string, options?: WidgetScaffoldingOptions): Promise<void>;

```

Reply:
widgetConfig is an object of related params, which are consumed by Dev Portal, deployConfig are again used for deployment and miscConfig are rest of various value, for there's just one but to future proof, I would keep it as an object. I can rename it to options if you would like to.
I don't think it makes sense to give default values to any of them.